docs/user_guide.md

```diff
@@ -719,6 +810,237 @@ If you are a project admin, you can disable a Robot Account by clicking "Disable
+
+|Event|Webhook Event Type|Contents of Notification|
+|---|---|---|
+|Push image to registry|`IMAGE PUSH`|Repository namespace name, repository name, resource URL, tags, manifest digest, image name, push time timestamp, username of user who pushed image|
+|Pull manifest from registry|`IMAGE PULL`|Repository namespace name, repository name, manifest digest, image name, pull time timestamp, username of user who pulled image|
+|Delete manifest from registry|`IMAGE DELETE`|Repository namespace name, repository name, manifest digest, image name, image size, delete time timestamp, username of user who deleted image|
+|Upload Helm chart to registry|`CHART PUSH`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of push, username of user who uploaded chart|
+|Download Helm chart from registry|`CHART PULL`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of push, username of user who pulled chart|
+|Delete Helm chart from registry|`CHART DELETE`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of delete, username of user who deleted chart|
+|Image scan completed|`IMAGE SCAN COMPLETED`|Repository namespace name, repository name, tag scanned, image name, number of critical issues, number of major issues, number of minor issues, last scan status, scan completion time timestamp, vulnerability information (CVE ID, description, link to CVE, criticality, URL for any fix), username of user who performed scan|
+|Image scan failed|`IMAGE SCAN FAILED`|Repository namespace name, repository name, tag scanned, image name, error that occurred, username of user who performed scan|
+
+#### JSON Payload Format
+
+The webhook notification is delivered in JSON format. The following example shows the JSON notification for a push image event:
+
+```
+{
+ "event_type": "pushImage"
+    "events": [
+               {
+                "project": "prj",
+                "repo_name": "repo1",
+                "tag": "latest",
+                "full_name": "prj/repo1",
+                "trigger_time": 158322233213,
+                "image_id": "9e2c9d5f44efbb6ee83aecd17a120c513047d289d142ec5738c9f02f9b24ad07",
+                "project_type": "Private"
+               }
+             ]
+}
+```
+
+### Webhook Endpoint Recommendations
+
+The endpoint that receives the webhook should ideally meet the following requirements:
+
+- Have a webhook listener that is capable of interpreting the payload and acting upon the information it contains. For example, running a shell script.
+- Support queues and retry configurations, so that failed webhook notifications are queued and retried until the request succeeds or number of retries exceeds the configured maximum.

```

Comment:
This is not a requirement of the endpoint, the endpoint just catches the payload.  This is just a general requirement of the webhook mechanism on the harbor side, support if it wasn't clear from the prd. @stuclem

Reply:
@xaleeks  I deleted that bullet and merged the previous bullet into the first sentence.